platform/security/sts/security-sts-attributequeryclaimshandler/src/main/java/org/codice/ddf/security/claims/attributequery/AttributeQueryClaimsHandler.java

```diff
@@ -0,0 +1,271 @@
+
+        ProcessedClaimCollection claimCollection = null;
+
+        try {
+            if (!StringUtils.isEmpty(username)) {
+                claimCollection = retrieveExternalClaimValues(username);
+            }
+        } catch (URISyntaxException e) {
+            LOGGER.error(ERROR_RETRIEVING_ATTRIBUTES, externalAttributeStoreUrl, username, e);
+            SecurityLogger.logError(
+                    String.format(ERROR_RETRIEVING_ATTRIBUTES_SEC_LOG, externalAttributeStoreUrl,
+                            username), e);
+        }
+
+        return (claimCollection != null) ? claimCollection : new ProcessedClaimCollection();
+    }
+
+    /**
+     * Retrieves claims from the external attribute store.
+     *
+     * @param username of the user.
+     * @return The collection of claims.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection retrieveExternalClaimValues(String username)
+            throws URISyntaxException {
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+
+        ProcessedClaimCollection securityClaimCollection = this.getAttributes(username);
+
+        //If security claim collection came back null, return a null claim collection.
+        if (securityClaimCollection == null || securityClaimCollection.size() == 0) {
+            return null;
+        }
+
+        claimCollection.addAll(securityClaimCollection);
+
+        return claimCollection;
+    }
+
+    /**
+     * Parses the CN from the user's DN.
+     *
+     * @param principal of the user.
+     * @return CN of the user.
+     */
+    protected String getUsername(Principal principal) {
+        String cn = null;
+        StringTokenizer st = new StringTokenizer(principal.getName(), ",");
+        while (st.hasMoreElements()) {
+            String[] strArr = st.nextToken().split("=");
+            if (strArr.length > 1 && strArr[0].equalsIgnoreCase("cn")) {
+                cn = strArr[1];
+                break;
+            }
+        }
+        return cn;
+    }
+
+    /**
+     * Gets the attributes for the supplied user from the external attribute store.
+     *
+     * @param username used for the request.
+     * @return The collection of attributes retrieved from the external attribute store.
+     * @throws URISyntaxException
+     */
+    protected ProcessedClaimCollection getAttributes(String username) throws URISyntaxException {
+        ProcessedClaimCollection claimCollection = new ProcessedClaimCollection();
+
+        LOGGER.debug("Sending AttributeQuery Request.");
+
+        Assertion assertion = null;
+        try {
+            assertion = createAttributeQueryClient(simpleSign, externalAttributeStoreUrl, issuer,
+                    destination).retrieveResponse(username);
+        } catch (AttributeQueryException ex) {
+            LOGGER.error("Error occurred in AttributeQueryClient, did not retrieve response.", ex);
+        }
+
+        if (assertion != null) {
```

Comment:
Actually, it would be null in the case if the Response object came back with a Status Code other than Success. An error would be reported and a null assertion would be returned.